src/transport/HolepunchTransport.ts

```diff
@@ -31,6 +33,15 @@ class HolepunchTransport extends ATransport {
     }
     send(serializedRPC: string): void {
         console.log("SENDING RPC", serializedRPC);
+
+        if (outboundRateLimiter) {
+            const dataSizeBytes = Buffer.byteLength(serializedRPC, "utf8");
```

Comment:
Would consider extracting this and writing it only once in the base class for all transports and then the actual concrete transports be maybe private methods that are called by the base class if they can write, so we don't end up copy pasting this in every transport we implement 

```typescript
ATransport{
  abstract _send()
  send(){
    rateLimit
    // more shared code here
    this._send()
  }
}
```